alf/algorithms/mbrl_algorithm.py

```diff
@@ -155,21 +155,30 @@ def _predict_next_step(self, time_step, state):
             next_state = state._replace(dynamics=dynamics_step.state)
         return next_time_step, next_state
 
-    def _calc_step_reward(self, obs, action):
-        reward = self._reward_module.compute_reward(obs, action)
+    def _calc_step_reward(self, obs, action, state: MbrlState):
+        """Calculate the step reward based on the given observation, action
+            and state.
+        Args:
+            obs (Tensor): observation
+            action (Tensor): action
+            state: state for reward calculation
+        Returns:
+            reward (Tensor): compuated reward for the given input
+        """
+        reward = self._reward_module.compute_reward(obs, action, state.reward)
```

Comment:
state.reward needs to be updated using the new state after compute_reward.

Reply:
Updated.